# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: third_party/rust/ron/tests/401_raw_identifier.rs
# Commit: 3f167f8b3137
# Full Hash: 3f167f8b313749f50f135e189e5854bef7ac0c73
# Author: Erich Gubler <erichdongubler@gmail.com>
# Date: 2025-10-03 18:08 +0000
# Regressor Bug: 1991551
# File Overlap Count: 23
# Overlapping Files: third_party/rust/wgpu-hal/Cargo.toml, supply-chain/audits.toml, third_party/rust/naga/src/back/spv/writer.rs, third_party/rust/naga/src/back/msl/writer.rs, third_party/rust/wgpu-types/src/lib.rs
# Description:
#   Bug 1991551 - build(webgpu): update WGPU to 8c4aebc0c1b75e3f7412dfc180745a0717fc30ec r=webgpu-reviewers,supply-chain-reviewers,nical
# ==============================================================================

diff -r 823439f307a0 -r 3f167f8b3137 third_party/rust/ron/tests/401_raw_identifier.rs
--- a/third_party/rust/ron/tests/401_raw_identifier.rs	Fri Oct 03 18:01:30 2025 +0000
+++ b/third_party/rust/ron/tests/401_raw_identifier.rs	Fri Oct 03 18:08:29 2025 +0000
@@ -1,4 +1,4 @@
-use ron::error::{Error, Position, SpannedError};
+use ron::error::{Position, Span, SpannedError};
 use serde::{Deserialize, Serialize};
 
 #[derive(Serialize, Deserialize, Debug)]
@@ -32,24 +32,27 @@
     );
     assert_eq!(
         ser,
-        Err(Error::InvalidIdentifier(String::from("Hello World")))
+        Err(ron::Error::InvalidIdentifier(String::from("Hello World")))
     );
 
     let ser = ron::ser::to_string_pretty(
         &EmptyStruct,
         ron::ser::PrettyConfig::default().struct_names(true),
     );
-    assert_eq!(ser, Err(Error::InvalidIdentifier(String::from(""))));
+    assert_eq!(ser, Err(ron::Error::InvalidIdentifier(String::from(""))));
 
     let de = ron::from_str::<InvalidStruct>("Hello World").unwrap_err();
     assert_eq!(
         de,
         SpannedError {
-            code: Error::ExpectedDifferentStructName {
+            code: ron::Error::ExpectedDifferentStructName {
                 expected: "Hello World",
                 found: String::from("Hello"),
             },
-            position: Position { line: 1, col: 6 },
+            span: Span {
+                start: Position { line: 1, col: 1 },
+                end: Position { line: 1, col: 6 },
+            }
         }
     );
 
@@ -57,8 +60,11 @@
     assert_eq!(
         de,
         SpannedError {
-            code: Error::ExpectedUnit,
-            position: Position { line: 1, col: 1 },
+            code: ron::Error::ExpectedUnit,
+            span: Span {
+                start: Position { line: 1, col: 1 },
+                end: Position { line: 1, col: 1 },
+            }
         }
     );
 
@@ -72,8 +78,11 @@
     assert_eq!(
         de,
         SpannedError {
-            code: Error::ExpectedNamedStructLike("Hello+World"),
-            position: Position { line: 1, col: 1 },
+            code: ron::Error::ExpectedNamedStructLike("Hello+World"),
+            span: Span {
+                start: Position { line: 1, col: 1 },
+                end: Position { line: 1, col: 1 },
+            }
         },
     );
 
@@ -81,8 +90,11 @@
     assert_eq!(
         de,
         SpannedError {
-            code: Error::ExpectedNamedStructLike("Hello+World"),
-            position: Position { line: 1, col: 1 },
+            code: ron::Error::ExpectedNamedStructLike("Hello+World"),
+            span: Span {
+                start: Position { line: 1, col: 1 },
+                end: Position { line: 1, col: 1 },
+            }
         },
     );
 
@@ -90,8 +102,11 @@
     assert_eq!(
         de,
         SpannedError {
-            code: Error::SuggestRawIdentifier(String::from("Hello+World")),
-            position: Position { line: 1, col: 1 },
+            code: ron::Error::SuggestRawIdentifier(String::from("Hello+World")),
+            span: Span {
+                start: Position { line: 1, col: 1 },
+                end: Position { line: 1, col: 1 },
+            }
         }
     );
 
@@ -104,8 +119,11 @@
     assert_eq!(
         de,
         SpannedError {
-            code: Error::SuggestRawIdentifier(String::from("ab.cd-ef")),
-            position: Position { line: 2, col: 9 },
+            code: ron::Error::SuggestRawIdentifier(String::from("ab.cd-ef")),
+            span: Span {
+                start: Position { line: 1, col: 15 },
+                end: Position { line: 2, col: 9 },
+            }
         }
     );
 
@@ -118,8 +136,11 @@
     assert_eq!(
         de,
         SpannedError {
-            code: Error::SuggestRawIdentifier(String::from("rab.cd-ef")),
-            position: Position { line: 2, col: 9 },
+            code: ron::Error::SuggestRawIdentifier(String::from("rab.cd-ef")),
+            span: Span {
+                start: Position { line: 1, col: 15 },
+                end: Position { line: 2, col: 9 },
+            }
         }
     );
 
@@ -132,12 +153,15 @@
     assert_eq!(
         de,
         SpannedError {
-            code: Error::NoSuchStructField {
+            code: ron::Error::NoSuchStructField {
                 expected: &["ab.cd-ef", "really_not_raw"],
                 found: String::from("ab.cd+ef"),
                 outer: Some(String::from("Hello+World")),
             },
-            position: Position { line: 2, col: 19 },
+            span: Span {
+                start: Position { line: 2, col: 11 },
+                end: Position { line: 2, col: 19 },
+            }
         }
     );
 
@@ -145,8 +169,11 @@
     assert_eq!(
         de,
         SpannedError {
-            code: Error::SuggestRawIdentifier(String::from("Hello-World")),
-            position: Position { line: 1, col: 1 },
+            code: ron::Error::SuggestRawIdentifier(String::from("Hello-World")),
+            span: Span {
+                start: Position { line: 1, col: 1 },
+                end: Position { line: 1, col: 1 },
+            }
         }
     );
 
@@ -154,19 +181,22 @@
     assert_eq!(
         de,
         SpannedError {
-            code: Error::NoSuchEnumVariant {
+            code: ron::Error::NoSuchEnumVariant {
                 expected: &["Hello-World"],
                 found: String::from("Hello+World"),
                 outer: Some(String::from("RawEnum")),
             },
-            position: Position { line: 1, col: 14 },
+            span: Span {
+                start: Position { line: 1, col: 3 },
+                end: Position { line: 1, col: 14 },
+            }
         }
     );
 
     let de = ron::from_str::<EmptyStruct>("r#+").unwrap_err();
     assert_eq!(
         format!("{}", de),
-        r#"1:4: Expected only opening `(`, no name, for un-nameable struct"#,
+        r#"1:3-1:4: Expected only opening `(`, no name, for un-nameable struct"#,
     );
 }
 